onionshare_gui/tor_connection_dialog.py

```diff
@@ -86,6 +86,7 @@ def _connected_to_tor(self):
     def _canceled_connecting_to_tor(self):
         common.log('TorConnectionDialog', '_canceled_connecting_to_tor')
         self.active = False
+        self.onion.cleanup()
```

Comment:
Sort of a mini-bug fixed here. When the Tor Connection dialog was canceled, the actual Onion process continued to run in the background. In theory this could lead to multiple background tor connections if the user later clicked Save under the SettingsDialog.. or at worst, it meant that Onion.is_authenticated() would evaluate to True instead of False, which affected the fixes being implemented in this PR since they heavily interrogate that function.

Reply:
If you open Onionshare with use Tor bundled with onionshare enabled.
Make sure Torbrowser is closed and enable "Attempt to automatically configure with Tor Browser" and click save, your will get a extra "Would you like to open OnionShare settings to troubleshoot connecting to Tor?" popup even though settings is all ready open. Clicking the exit settings button will open 2 settings dialogs.

Reply:
Cool, so I officially give up on this one, it's just too complex for my skill. Thanks for testing.

Gonna close this but leave my branch so someone else can maybe see how I tried to fix it.